s3 gateway closed connection without right answer to client

I tried to load file AWSCLIV2.msi (~30Mb) to s3 gateway following this command:
```shell
$ aws --no-verify-ssl s3api put-object --bucket new-bucket-1508 --key new-object1 --body c:\TEMP\AWSCLIV2.msi --endpoint-url http://172.26.160.159:8084
```
But instead of answer "access denied", s3 gateway closed connection:
`Connection was closed before we received a valid response from endpoint URL: "http://172.26.160.159:8084/new-bucket-1508/new-object1"`

**Steps to reproduce:**
1. Create s3 creredentials:
```shell
$ ./neofs-s3-authmate-linux-amd64 issue-secret --wallet /home/anikeev/neofs/tatlin-object-sber-tfstate/performance/configuration/wallets/node3-s3.json --peer 172.26.160.159:8080 --gate-public-key 022207a98e2d33e2e8a53f1205d8bedac69c04cb3622b0ed318800840b49baa10b
```
2. Put s3 keys to aws
3. Create bucket:
```shell
$ aws --no-verify-ssl s3api create-bucket --bucket new-bucket-1508 --endpoint-url http://172.26.160.159:8084
```
4. Load object to bucket:
```shell
$ aws --no-verify-ssl s3api put-object --bucket new-bucket-1508 --key new-object1 --body c:\TEMP\AWSCLIV2.msi --endpoint-url http://172.26.160.159:8084
```

**s3 gateway version:**
NeoFS S3 Gateway
Version: v0.23.0
GoVersion: go1.18.4

**Part of log of this event:**
```
Aug 16 08:07:17 az neogo-morph-cn[330]: 2022-08-16T08:07:17.465Z        INFO        persisted to disk        {"blocks": 1, "keys": 40, "headerHeight": 36264, "blockHeight": 36264, "took": "29.819976ms"}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.424Z        INFO        received PrepareRequest        {"validator": 1, "tx": 0}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.424Z        INFO        sending PrepareResponse        {"height": 36265, "view": 0}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.425Z        INFO        received PrepareResponse        {"validator": 2}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.426Z        INFO        sending Commit        {"height": 36265, "view": 0}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.426Z        INFO        received Commit        {"validator": 0}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.427Z        INFO        received Commit        {"validator": 2}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.427Z        INFO        approving block        {"height": 36265, "hash": "ddecbc541670227b0b7ba44306345d9e2a2d96333920d3fd29ca3e2bb2f6388a", "tx_count": 0, "merkle": "0000000000000000000000000000000000000000000000000000000000000000", "prev": "1e346b0a26a5880ab6f9121fea85fc12fd2fc562d49fcc3f598587e110103af1"}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.429Z        INFO        initializing dbft        {"height": 36266, "view": 0, "index": 3, "role": "Backup"}
Aug 16 08:07:25 az neogo-morph-cn[330]: 2022-08-16T08:07:25.454Z        INFO        persisted to disk        {"blocks": 1, "keys": 21, "headerHeight": 36265, "blockHeight": 36265, "took": "16.099368ms"}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.428Z        INFO        received PrepareRequest        {"validator": 2, "tx": 0}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.428Z        INFO        sending PrepareResponse        {"height": 36266, "view": 0}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.429Z        INFO        received PrepareResponse        {"validator": 0}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.429Z        INFO        sending Commit        {"height": 36266, "view": 0}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.429Z        INFO        received Commit        {"validator": 0}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.441Z        INFO        received PrepareResponse        {"validator": 1}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.441Z        INFO        received Commit        {"validator": 2}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.441Z        INFO        approving block        {"height": 36266, "hash": "8945b1d73b9b3374fb605b71c1b522d88626476874b8fe6fabdd4764588f9d42", "tx_count": 0, "merkle": "0000000000000000000000000000000000000000000000000000000000000000", "prev": "ddecbc541670227b0b7ba44306345d9e2a2d96333920d3fd29ca3e2bb2f6388a"}
Aug 16 08:07:33 az neogo-morph-cn[330]: 2022-08-16T08:07:33.442Z        INFO        initializing dbft        {"height": 36267, "view": 0, "index": 3, "role": "Primary"}
Aug 16 08:07:34 az neogo-morph-cn[330]: 2022-08-16T08:07:34.457Z        INFO        persisted to disk        {"blocks": 1, "keys": 19, "headerHeight": 36266, "blockHeight": 36266, "took": "15.552347ms"}
Aug 16 08:07:36 az neofs-node[837]: 2022-08-16T08:07:36.407Z        warn        engine/engine.go:44        could not get object from shard        {"shard_id": "YFbkULuXawC3Q3Cu2thFTa", "error count": 5, "error": "could not open blobovnicza 0/3/0: timeout"}
Aug 16 08:07:36 az neofs-node[837]: 2022-08-16T08:07:36.407Z        warn        engine/engine.go:44        meta info was present, but object is missing        {"shard_id": "YFbkULuXawC3Q3Cu2thFTa", "error count": 6, "error": "could not open blobovnicza 0/3/0: timeout", "address": "Aq3b9ziPiUX4j2jqa3o9NmSPWWtxFC7E1rc33EofCFed/47bqiCF2V1NYEZhfoPC5YsUVwTLY8qfnXaLZv4pnBbSy"}
Aug 16 08:07:36 az neofs-s3-gw[614]: 2022-08-16T08:07:36.461Z        debug        layer/object.go:706        error was transformed        {"request_id": "386d945e-d98a-429b-8252-23e5f9ecfe9e", "error": "access denied: access to operation OBJECT_PUT is denied by extended ACL check: denied by rule"}
Aug 16 08:07:36 az neofs-s3-gw[614]: 2022-08-16T08:07:36.461Z        error        handler/util.go:25        could not upload object        {"request_id": "386d945e-d98a-429b-8252-23e5f9ecfe9e", "method": "PutObject", "bucket_name": "new-bucket-1508", "object_name": "new-object4", "error": "AccessDenied: 403 => Access Denied."}
Aug 16 08:07:36 az neofs-s3-gw[614]: 2022-08-16T08:07:36.461Z        error        api/router.go:158        something went wrong        {"status": 403, "request_id": "386d945e-d98a-429b-8252-23e5f9ecfe9e", "method": "PutObject", "description": "Forbidden"}
Aug 16 08:07:37 az neofs-s3-gw[614]: 2022-08-16T08:07:37.228Z        debug        layer/object.go:706        error was transformed        {"request_id": "9ac32f5e-4dab-41c8-b022-944f2b63197d", "error": "access denied: access to operation OBJECT_PUT is denied by extended ACL check: denied by rule"}
Aug 16 08:07:37 az neofs-s3-gw[614]: 2022-08-16T08:07:37.228Z        error        handler/util.go:25        could not upload object        {"request_id": "9ac32f5e-4dab-41c8-b022-944f2b63197d", "method": "PutObject", "bucket_name": "new-bucket-1508", "object_name": "new-object4", "error": "AccessDenied: 403 => Access Denied."}
Aug 16 08:07:37 az neofs-s3-gw[614]: 2022-08-16T08:07:37.228Z        error        api/router.go:158        something went wrong        {"status": 403, "request_id": "9ac32f5e-4dab-41c8-b022-944f2b63197d", "method": "PutObject", "description": "Forbidden"}
Aug 16 08:07:38 az neofs-s3-gw[614]: 2022-08-16T08:07:38.407Z        debug        layer/object.go:706        error was transformed        {"request_id": "7ae89481-00f4-44ba-ad62-0598ad7089eb", "error": "access denied: access to operation OBJECT_PUT is denied by extended ACL check: denied by rule"}
Aug 16 08:07:38 az neofs-s3-gw[614]: 2022-08-16T08:07:38.407Z        error        handler/util.go:25        could not upload object        {"request_id": "7ae89481-00f4-44ba-ad62-0598ad7089eb", "method": "PutObject", "bucket_name": "new-bucket-1508", "object_name": "new-object4", "error": "AccessDenied: 403 => Access Denied."}
Aug 16 08:07:38 az neofs-s3-gw[614]: 2022-08-16T08:07:38.407Z        error        api/router.go:158        something went wrong        {"status": 403, "request_id": "7ae89481-00f4-44ba-ad62-0598ad7089eb", "method": "PutObject", "description": "Forbidden"}
```
**s3 config:**
```yaml
default_policy: REP 1 IN X CBF 1 SELECT 1 FROM * AS X
listen_address: 0.0.0.0:8084
logger:
  level: debug
max_clients_count: 600
max_clients_deadline: 60s
peers:
  '0':
    address: node1.neofs:8080
    priority: '1'
    weight: '1'
  '1':
    address: node2.neofs:8080
    priority: '2'
    weight: '0.25'
  '2':
    address: node3.neofs:8080
    priority: '2'
    weight: '0.25'
  '3':
    address: node4.neofs:8080
    priority: '2'
    weight: '0.25'
pool_error_threshold: 100
pprof:
  address: localhost:8085
  enabled: true
prometheus:
  address: localhost:8086
  enabled: true
resolve_order:
- nns
rpc_endpoint: http://node1.neofs:40332
tree:
  service: 172.26.160.159:8080
wallet: /etc/neofs/s3/wallet.json
```

--- a/api/handler/put.go
+++ b/api/handler/put.go
@@ -248,7 +248,9 @@ func (h *handler) PutObjectHandler(w http.ResponseWriter, r *http.Request) {
 
 	info, err := h.obj.PutObject(r.Context(), params)
 	if err != nil {
-		h.logAndSendError(w, "could not upload object", reqInfo, err)
+		_, err2 := io.Copy(io.Discard, r.Body)
+		err3 := r.Body.Close()
+		h.logAndSendError(w, "could not upload object", reqInfo, err, zap.Errors("body close errors", []error{err2, err3}))
 		return
 	}
 

